c10/util/C++17.h

```diff
@@ -45,14 +45,7 @@ constexpr bool is_pod_v = is_pod<T>::value;
 
 namespace guts {
 
-#if defined(__cpp_lib_apply) && !defined(__CUDA_ARCH__) && !defined(__HIP__)
-
-template <class F, class Tuple>
-C10_HOST_DEVICE inline constexpr decltype(auto) apply(F&& f, Tuple&& t) {
```

Comment:
The std version is missing the HOST_DEVICE macro, can we get rid of it?

Reply:
It builds so that `HOST_DEVICE` may be not needed

Reply:
It's need for ROCM apparently

Reply:
OK, let's revert some changes.